### PR TITLE
Fix missing search results

### PR DESCRIPTION
**Pull request by woodwoerk**
When typing the same search query twice, no results were displayed for the second query.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206486884287691

### CODE REVIEW

File: components/common/NavbarSearch.js
Comment (woodwoerk) [inline components/common/NavbarSearch.js:1]: Note: Everything except the loading status changes (lines 267-284) is just refactoring. I thought the issue may have been due to using React hooks in the `WithSearch` callback so refactored that callback to a separate component. Didn't turn out to be the issue but at least React isn't complaining at us now.

File: common/search.ts [hunk 1]
```diff
@@ -46,20 +45,14 @@ const GET_AUTOCOMPLETE_RESULTS = gql`
...
     this.plan = plan;
   }
 
   async onSearch(opts, queryConfig) {
-    // Not implemented, stub here just to silence the warning about
-    // unimplemented method.
-    return { results: [] };
-  }
-
-  async onAutocomplete(opts, queryConfig) {
```
Comment (woodwoerk) [inline common/search.ts:62]: @terotik This was a pain to debug but it turns out the issue was an `onSearch` and `onAutocomplete` race condition. `onSearch` would resolve after `onAutocomplete` when querying searches for a second time, therefor overriding the list of results with an empty array. Any idea why we also used `onAutocomplete` here? Just using `onSearch` seems to do the trick.

Comment (terotik) [reply common/search.ts:62]: I'm getting `Uncaught No onAutocomplete handler provided and no Connector provided. You must configure one or the other.` warning on console running this locally. Could that be the reason? Not sure if it's a problem if it works though...

Comment (woodwoerk) [reply common/search.ts:62]: That might be a problem in production builds, I'll take another look

Comment (woodwoerk) [reply common/search.ts:62]: @terotik Can you test this locally again? I disabled autocomplete [here](https://github.com/kausaltech/kausal-watch-ui/pull/272/commits/83ae1463aee0994dd386ad062d7c3cf28296d851)

**Review by terotik — APPROVED** [review]

**Pull request merged**